src/syscheckd/syscheck_audit.c

```diff
@@ -573,6 +573,37 @@ char *gen_audit_path(char *cwd, char *path0, char *path1) {
     return gen_path;
 }
 
+void get_parent_process_info(char *ppid, char **parent_name, char **parent_cwd) {
+
+    char *slinkexe = NULL;
+    char *slinkcwd = NULL;
+    int tam_slink = strlen(ppid) + 11;
+    int tam_ppname = 0;
+    int tam_pcwd = 0;
+
+    os_malloc(tam_slink, slinkexe);
+    os_malloc(tam_slink, slinkcwd);
+
+    snprintf(slinkexe, tam_slink, "/proc/%s/exe", ppid);
+    snprintf(slinkcwd, tam_slink, "/proc/%s/cwd", ppid);
+
+    if(tam_ppname = readlink(slinkexe, *parent_name, OS_FLSIZE), tam_ppname < 0) {
+        merror("Failure to obtain the name of the process: '%s'. Error: %s", ppid, strerror(errno));
+        parent_name[0][0] = '\0';
+    } else {
+        parent_name[0][tam_ppname] = '\0';
+    }
+
+    if(tam_pcwd = readlink(slinkcwd, *parent_cwd, OS_FLSIZE), tam_pcwd < 0) {
+        merror("Failure to obtain the cwd of the process: '%s'. Error: %s", ppid, strerror(errno));
```

Comment:
Please change this error message to a debug1. In case we can't get the CWD or the PPN we should still be alert, but without those fields. This does not imply an unrecoverable error.